 <div><h4><pre><strong>Name of experiment = CREATE A MODEL USING PYTHON FROM FLOWCHART FOR GIVEN NUMERICAL<strong></pre></h4></div>

In [1]:
# Function defining the ODE
def f(x, y):
    return x + y**2

# 4th-order Runge-Kutta method
def runge_kutta_4th_order(x, y, h):
    k1 = h * f(x, y)
    k2 = h * f(x + h/2, y + k1/2)
    k3 = h * f(x + h/2, y + k2/2)
    k4 = h * f(x + h, y + k3)
    
    y_next = y + (k1 + 2*k2 + 2*k3 + k4) / 6
    return y_next

# Given initial values and conditions
x_0 = 0
y_0 = 1
h = 0.1
x_target = 0.2

# Number of steps required
num_steps = int((x_target - x_0) / h)

# Applying the Runge-Kutta method using a for loop
for _ in range(num_steps):
    y_0 = runge_kutta_4th_order(x_0, y_0, h)
    x_0 += h

# Print the result
print("Approximate value of y at x = 0.2:", y_0)


Approximate value of y at x = 0.2: 1.2735625426752228


 <div><h4><pre><strong>
Conclusion::
The code uses the 4th-order Runge-Kutta method to numerically solve an ordinary differential equation.
Approximate value of y at x = 0.2: 1.2735625426752228
<strong></pre></h4></div>